01-30-shield-blocking/actors/player/Player.gd

```diff
@@ -0,0 +1,78 @@
+extends KinematicBody2D
+class_name Player
+
+onready var invencible_timer : Timer = $InvencibleTimer
+onready var animated_sprite : AnimatedSprite = $AnimatedSprite
+onready var animation_player : AnimationPlayer = $AnimationPlayer
+onready var shield : Node2D = $Shield
+
+export var move_speed : = 150.0
+export var jump_force : = 200.0
+export var gravity : = 25.0
+
+var velocity : = Vector2()
+var blocking : = false
+
+
+func _process(delta: float) -> void:
```

Comment:
Don't manage input in _process: use _unhandled_input except if you really can't. Checking on input constantly like that instead of processing individual events is a source of bugs